Cpp/fostgres/datum.cpp

```diff
@@ -15,6 +16,39 @@
 
 namespace {
     const fostlib::json c_file("file");
+    constexpr f5::u8view redacted{"XXXXREDACTED"};
+
+    /// ASCII based case insensitive comparison for equality which performs
+    /// the same check as the header `map` does for the key names
+    template<typename L, typename R>
+    inline bool ieq(L &&l, R &&r) {
+        fostlib::detail::ascii_iless less{};
+        return not(less(l, r) || less(r, l));
+    }
+
+    fostlib::nullable<fostlib::json>
+            header_redaction(std::size_t const pathlen, fostlib::json val) {
+        if (pathlen == 2) {
+            auto const parts = fostlib::partition(
+                    fostlib::coerce<fostlib::string>(val), " ");
+            if (parts.first == "Bearer" && parts.second
+                && parts.second->startswith("ey")) {
+                auto end = parts.second->find('.');
+                if (end != f5::u8view::npos) {
+                    end = parts.second->find('.', end + 1);
+                    if (end != f5::u8view::npos) {
+                        return fostlib::json{
+                                "Bearer " + parts.second->substr(0, end + 1)
```

Comment:
Do you want us to store `Bearer ` to db?

Reply:
The solution here has to be a general one suitable for a wide variety of use cases. If we tailor this to the absolute minimum that you require then it becomes less useful for other use cases. I assume that you stripping the "Bearer " from the start of the string if you don't are about it isn't a problem for you :-) You may want to check that it actually is Bearer though...